File: solutions/csharp/resistor-color-duo/1/ResistorColorDuo.cs
using System;

public static class ResistorColorDuo
{
    static readonly string[] BAND_COLORS = new[] { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};
    
    public static int Value(string[] colors)
        => Array.IndexOf(BAND_COLORS, colors[0]) * 10 
         + Array.IndexOf(BAND_COLORS, colors[1]);
}

File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export function decodedValue(bands: string[]) {
  return decode_band(bands[0]) * 10 + decode_band(bands[1]);
}

enum BandValue {
  black,
  brown,
  red,
  orange,
  yellow,
  green,
  blue,
  violet,
  grey,
  white 
}

function decode_band(color: string): number {
  let colorKey = color as keyof typeof BandValue;
  return BandValue[colorKey];
}